chain/client/src/chunk_inclusion_tracker.rs

```diff
@@ -56,35 +89,78 @@ impl ChunkInclusionTracker {
         self.banned_chunk_producers.put((epoch_id, account_id), ());
     }
 
+    /// Update signatures in chunk_info
+    pub fn prepare_chunk_headers_ready_for_inclusion(
+        &mut self,
+        prev_block_hash: &CryptoHash,
+        chunk_validator: &mut ChunkValidator,
+    ) -> Result<(), Error> {
+        let Some(entry) = self.prev_block_to_chunk_hash_ready.get(prev_block_hash) else {
+            return Ok(());
+        };
+
+        for chunk_hash in entry.values() {
+            let chunk_info = self.chunk_hash_to_chunk_info.get_mut(chunk_hash).unwrap();
+            chunk_info.signatures =
+                chunk_validator.get_chunk_endorsement_signatures(&chunk_info.chunk_header)?;
+        }
+        Ok(())
+    }
+
+    fn is_banned(&self, epoch_id: &EpochId, chunk_info: &ChunkInfo) -> bool {
+        let banned = self
+            .banned_chunk_producers
+            .contains(&(epoch_id.clone(), chunk_info.chunk_producer.clone()));
+        if banned {
+            tracing::warn!(
+                target: "client",
+                chunk_hash = ?chunk_info.chunk_header.chunk_hash(),
+                chunk_producer = ?chunk_info.chunk_producer,
+                "Not including chunk from a banned validator");
+            metrics::CHUNK_DROPPED_BECAUSE_OF_BANNED_CHUNK_PRODUCER.inc();
+        }
+        banned
+    }
+
+    fn has_chunk_endorsements(&self, chunk_info: &ChunkInfo) -> bool {
+        let has_chunk_endorsements = chunk_info.signatures.is_some();
+        if !has_chunk_endorsements {
+            tracing::warn!(
+                target: "client",
+                chunk_hash = ?chunk_info.chunk_header.chunk_hash(),
+                chunk_producer = ?chunk_info.chunk_producer,
+                "Not including chunk because of insufficient chunk endorsements");
+        }
+        // TODO(shreyan): return has_chunk_endorsements here after fixing testing infra
```

Comment:
As of this PR due to some major complexities, the tests are not passing. For now I'm ignoring `has_chunk_endorsements` and including chunks that don't have chunk endorsements as well.

Subsequent PRs that fix the test would uncomment this.